### PR TITLE
Fix map_headers when headers have the same prefix

### LINKED ISSUE #1450: Cannot map datatable headers if one header is a prefix of another
**Describe the bug**

An error is raised for `map_headers` on a `DataTable` if one header value is a prefix of another header value.

**To Reproduce**

```ruby
DataTable.from([["ant", "anteater"], [1, 2]]).map_headers { |h| h.capitalize }.hashes
# RuntimeError Exception: 2 headers matched "ant": ["ant", "anteater"]
```

**Expected behavior**
`[{"Ant"=>1, "Anteater"=>2}]`

**Your Environment**

Ruby 2.6.6p146
Cucumber 4.1.0

**Additional context**

Appears to be due to this PR: https://github.com/cucumber/cucumber-ruby/commit/7b9825084e883aa139d220f1f482eb84a3c67a3b#diff-0466a7e3105b8e1a9e19cb8b0750f74fR510

### PR DESCRIPTION
**Pull request by valerianb**
# Description

Fixes #1450 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] Tests have been added for any changes to behaviour of the code
- [x] New and existing tests are passing locally and on CI
- [x] `bundle exec rubocop` reports no offenses
- [x] RDoc comments have been updated
- [x] CHANGELOG.md has been updated

### CODE REVIEW

**Review by luke-hill — APPROVED** [review]

**Comment by luke-hill** [comment]
RtM pending CI.

**Review by aurelien-reeves — APPROVED** [review]
Looks good, thanks!

File: lib/cucumber/multiline_argument/data_table.rb [hunk 1]
```diff
@@ -492,7 +492,7 @@ def convert_headers! # :nodoc:
         end
 
         @header_mappings.each_pair do |pre, post|
-          mapped_cells = header_cells.reject { |cell| cell.value.match(pre).nil? }
+          mapped_cells = header_cells.reject { |cell| pre.is_a?(Regexp) ? cell.value.match(pre).nil? : cell.value != pre }
```
Comment (olleolleolle) [inline lib/cucumber/multiline_argument/data_table.rb:495]: Is this an opportunity to use #match? ?

Comment (valerianb) [reply lib/cucumber/multiline_argument/data_table.rb]: Done :)

**Comment by aslakhellesoy** [comment]
Hi @valerianb,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Comment by aurelien-reeves** [comment]
Great work, thanks for your contribution @valerianb 🙏

**Comment by valerianb** [comment]
Thanks all for your reviews and welcoming me here.

**Pull request merged**